### PR TITLE
EM-140: Reorder documents in a given collection.

### PR DESCRIPTION
**Pull request by esune**
This review contains:
* changes to the collections controller and html to be able to drag&drop documents within and across document groups.
* styling tweaks (thanks @jeznorth) to ensure consistency with the document manager, as well as a good user experience when rearranging documents.
* the old document reorder code (that was not being used) has also been removed.

### CODE REVIEW

File: modules/collections/client/directives/collections.reorder.js [hunk 1]
```diff
@@ -1,169 +0,0 @@
-"use strict";
```
Comment (esune) [inline modules/collections/client/directives/collections.reorder.js]: This is all old and finally gone.

File: modules/collections/client/views/collections-reorder-content.html [hunk 2]
```diff
@@ -1,34 +0,0 @@
-<div class="dnd-drag-target-top" dnd-scroll-area dnd-region="top" dnd-scroll-container="collectionList"></div>
```
Comment (esune) [inline modules/collections/client/views/collections-reorder-content.html]: Old, unused, gone.

File: modules/collections/client/views/modal-collections-reorder.html [hunk 3]
```diff
@@ -1,23 +0,0 @@
-
```
Comment (esune) [inline modules/collections/client/views/modal-collections-reorder.html]: Old, unused, gone.

File: modules/collections/server/controllers/collections.controller.js [hunk 4]
```diff
@@ -136,60 +136,6 @@ module.exports = DBModel.extend({
     });
   },
 
-  sortOtherDocuments: function (collectionId, idList) {
```
Comment (esune) [inline modules/collections/server/controllers/collections.controller.js]: These functions used the old sorting logic, so they have been removed.

File: modules/collections/server/controllers/collections.controller.js [hunk 5]
```diff
@@ -198,49 +144,43 @@ module.exports = DBModel.extend({
...
-      if (!_.find(docLists[targetDocType], function(cd) { return cd.document._id.equals(documentId); })) {
 
-        // remove from the other document list, if it is there
-        var inWatchDocList = _.filter(docLists[watchDocListType], function(cd) { return cd.document._id.equals(documentId); });
-        _.each(inWatchDocList, function(otherDoc){
-          self.removeDocument(collectionId, otherDoc.document._id, watchDocListType);
-        })
+      // find the database record for the document we are adding
+      var Document = new DocumentClass(self.opts);
+      return Document.findById(documentId).then(function(document) {
```
Comment (esune) [inline modules/collections/server/controllers/collections.controller.js]: This got cleaned up to support the new document drag&drop reordering, and was simplified in the process.

File: modules/collections/client/controllers/collections.controllers.js [hunk 6]
```diff
@@ -259,69 +255,77 @@ collectionModules.controller('CollectionEditCtrl',
...
+            $scope.saveCollectionDocuments(true);
+          })
+          .catch(function() {
+            AlertService.error('"' + $scope.collection.displayName + '" was not saved.', 4000, true);
+            self.goToEdit();
+          });
+      };
+
+      /*
+       * This objecect contains jQuery-sortable settings related to sortable document lists.
```
Comment (mjfortier) [inline modules/collections/client/controllers/collections.controllers.js]: Just for the sake of perfectionism, there's a slight typo in this comment :)

**Comment by NickPhura** [comment]
👍

**Comment by esune** [comment]
Well, it looks like @jeznorth found a little bug here. Closing this PR and re-submitting it once it is fixed.

**Pull request closed without merging**